service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EhrFolderEffectiveTimeTest.java

```diff
@@ -0,0 +1,176 @@
+    }
+
+    @Test
+    public void When_UpdatingEffectiveTimePeriodWithStartEarlier_Expect_StartUpdated() {
+        Period initialPeriod = getPeriod(String.format(ENCOUNTER_WITH_FULL_DATES, INITIAL_START, INITIAL_END));
+        Period newPeriod = getPeriod(String.format(ENCOUNTER_WITH_FULL_DATES, EARLIER_START, EARLIER_END));
+
+        EhrFolderEffectiveTime ehrFolderEffectiveTime = new EhrFolderEffectiveTime();
+        ehrFolderEffectiveTime.updateEffectiveTimePeriod(initialPeriod);
+        ehrFolderEffectiveTime.updateEffectiveTimePeriod(newPeriod);
+
+        assertThat(ehrFolderEffectiveTime.getEffectiveTimeLow().isPresent()).isTrue();
+        assertThat(ehrFolderEffectiveTime.getEffectiveTimeLow().get()).isEqualTo(EARLIER_START_DATE_HL7);
+
+        assertThat(ehrFolderEffectiveTime.getEffectiveTimeHigh().isPresent()).isTrue();
+        assertThat(ehrFolderEffectiveTime.getEffectiveTimeHigh().get()).isEqualTo(INITIAL_END_DATE_HL7);
+    }
+
+    @Test
+    public void When_UpdatingEffectiveTimePeriodWithEndLater_Expect_EndUpdated() {
+        Period initialPeriod = getPeriod(String.format(ENCOUNTER_WITH_FULL_DATES, INITIAL_START, INITIAL_END));
+        Period newPeriod = getPeriod(String.format(ENCOUNTER_WITH_FULL_DATES, LATER_START, LATER_END));
+
+        EhrFolderEffectiveTime ehrFolderEffectiveTime = new EhrFolderEffectiveTime();
+        ehrFolderEffectiveTime.updateEffectiveTimePeriod(initialPeriod);
+        ehrFolderEffectiveTime.updateEffectiveTimePeriod(newPeriod);
+
+        assertThat(ehrFolderEffectiveTime.getEffectiveTimeLow().isPresent()).isTrue();
+        assertThat(ehrFolderEffectiveTime.getEffectiveTimeLow().get()).isEqualTo(INITIAL_START_DATE_HL7);
+
+        assertThat(ehrFolderEffectiveTime.getEffectiveTimeHigh().isPresent()).isTrue();
+        assertThat(ehrFolderEffectiveTime.getEffectiveTimeHigh().get()).isEqualTo(LATER_END_DATE_HL7);
+    }
+
+    @Test
+    public void When_UpdatingEffectiveTimePeriodWithStartLaterInHl7Format_Expect_NoneUpdated() {
+        Period initialPeriod = getPeriod(String.format(ENCOUNTER_WITH_FULL_DATES, INITIAL_START, INITIAL_END));
+        Period newPeriod = getPeriod(String.format(ENCOUNTER_WITH_FULL_DATES, LATER_START, EARLIER_END));
+
+        EhrFolderEffectiveTime ehrFolderEffectiveTime = new EhrFolderEffectiveTime();
+        ehrFolderEffectiveTime.updateEffectiveTimePeriod(initialPeriod);
+        ehrFolderEffectiveTime.updateEffectiveTimePeriod(newPeriod);
+
+        assertThat(ehrFolderEffectiveTime.getEffectiveTimeLow().isPresent()).isTrue();
+        assertThat(ehrFolderEffectiveTime.getEffectiveTimeLow().get()).isEqualTo(INITIAL_START_DATE_HL7);
+
+        assertThat(ehrFolderEffectiveTime.getEffectiveTimeHigh().isPresent()).isTrue();
+        assertThat(ehrFolderEffectiveTime.getEffectiveTimeHigh().get()).isEqualTo(INITIAL_END_DATE_HL7);
+    }
+
+    @ParameterizedTest
+    @MethodSource("earlierLowDateInHl7PartDateFormat")
```

Comment:
This could just be a `@ValueSource` of strings to make it easier to read

```suggestion
    @ValueSource(strings = {"20180828103005", "20180828", "201808", "2018"})
```